plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -99,11 +81,24 @@ function auto_sizes_filter_image_tag( $content, array $parsed_block, WP_Block $b
 		 * @param string $size  The image size data.
 		 */
 		$filter = static function ( $sizes, $size ) use ( $block ) {
-			$id        = $block->attributes['id'] ?? 0;
-			$alignment = $block->attributes['align'] ?? '';
-			$width     = $block->attributes['width'] ?? '';
 
-			return auto_sizes_calculate_better_sizes( (int) $id, (string) $size, (string) $alignment, (string) $width );
+			$id            = $block->attributes['id'] ?? 0;
+			$alignment     = $block->attributes['align'] ?? '';
+			$width         = $block->attributes['width'] ?? '';
+			$max_alignment = $block->context['max_alignment'];
```

Comment:
Is this guaranteed to always be present? If not, maybe safer:

```suggestion
			$max_alignment = $block->context['max_alignment'] ?? '';
```

Reply:
We're explicitly setting it [here](https://github.com/WordPress/performance/blob/d3a60d0ab386fde7724f4f7e7de0fa6f102c6b6b/plugins/auto-sizes/includes/improve-calculate-sizes.php#L270), but we could fall back to an empty string if it gets unset for some reason.